nexus/sdk-python.md

```diff
@@ -0,0 +1,330 @@
+        return AnotherOutput(input.message)
+ 
+```
+
+For comparison, here’s the “fully manual” style again:
+
+```python
+@nexusrpc.handler.service(interface=interface.MyNexusService)
+class MyNexusService:
+    # User can pass anything to service constructor at worker-start time
+
+    @nexusrpc.handler.operation
+    def echo(self) -> nexusrpc.handler.Operation[EchoInput, EchoOutput]:
+        return EchoOperation()  # user can pass anything to operation constructor at worker-start time
+        
+    @nexusrpc.handler.operation
+    def another_op(self) -> nexusrpc.handler.Operation[AnotherInput, AnotherOutput]:
+        return AnotherOperation()
+
+class EchoOperation:
+    async def echo(self, input: EchoInput, _: nexusrpc.handler.StartOperationOptions) -> EchoOutput:
+        return EchoOutput(input.message)
+    
+class AnotherOperation:
+    async def another_op(self, input: AnotherInput, _: nexusrpc.handler.StartOperationOptions) -> AnotherOutput:
+        return AnotherOutput(input.message)
+
+```
+
+Note that:
+
+- In both styles, the `start` handler has access to shared state created at Worker-start time.
+- In the “fully manual” case, this shared state (`self`) is an operation instance (the user may have chosen to also make the service instance accessible from operation verb handlers)
+- In the shorthand form, the shared state (`self`) is the service instance; a user using this definition style has no access to an operation instance, as in Go and Java.
+- In both cases, the signatures written by the user are non-magical and completely explicit; they’ve chosen to define the `start` handler as a method on an operation/service class respectively.
+- In the shorthand form, the decorator takes in the `start` handler method and returns a factory method `(Service) -> Operation` (that factory method takes care of creating an operation with the specified start method). Thus at run-time, the method on their service has a signature that differs from what they wrote under the decorator. Decorators in Python are used to effect fairly radical transformations: for example `@property`, or `@contextlib.contextmanager` in the standard library. But the main point in defense of the radicalness of this decorator transformation is that the user is not expected to call this method: it is called by the framework.
+- The decorator propagates type annotations on the start method defined on the service to become run-time type annotation objects on the methods defined on the operation [[nexus-sdk PR](https://github.com/nexus-rpc/sdk-python/blob/e318efeb5b615a7dab916af1c5f01fe97ffff841/src/nexusrpc/handler.py#L294), [sdk-python PR](https://github.com/temporalio/sdk-python/blob/08430098aac4d92c3744b1245c5585819abadad3/temporalio/nexus/handler.py#L155-L169)]. This allows, e.g. a Nexus worker to instantiate the correct Python type from payloads received over the wire.
+
+**[ALTERNATIVES CONSIDERED]**
```

Comment:
Not sure this is an alternative. Isn't what is below effectively what `@nexusrpc.handler.sync_operation` is sugar for?